packages/core/admin/admin/src/utils/unique-admin-hash.js

```diff
@@ -0,0 +1,13 @@
+const hash = require('hash.js');
+
+const hashAdminUserEmail = (payload) => {
+  try {
+    const adminUserEmailHash = hash.sha256().update(payload.email).digest('hex');
+
+    return adminUserEmailHash;
+  } catch (error) {
+    return '';
+  }
+};
+
+module.exports = hashAdminUserEmail;
```

Comment:
Modern browsers have a crypto API build it that we could utilize for that, instead of including quite a big dependency (5kB): https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest

What do you think about using this standard sha256 method, which then allows to backend to use a node API?

Reply:
The support for that is actually quite good. If you want to be super-secure you could check for the feature before using it, but I think with the support Matrix like that you won't run into problems ...

<img width="781" alt="Screenshot 2022-10-25 at 16 41 04" src="https://user-images.githubusercontent.com/2244375/197804541-f8dfa375-a281-4edc-8c46-b128cf6b0407.png">

Reply:
Okay, sold 👍

Reply:
Great thanks. And sorry for being so late with the feedback - I just didn't know you were still working on it ...

Reply:
Tracker had to be also configured to accept all these changes 👍

Reply:
But I really appreciate the feedback, thank you 😊

Reply:
@gu-stav have you implemented web crypto stuff previously? I'm looking at it and there are some bits I'm not sure about.
Firstly, the code looks a bit more complex. Conversions to a hex string is not looking pretty in particular.
Secondly, it only runs in HTTPS. But even from experience, when trying to set up Strapi in development on a VPS or something, misconfigs are normal and people end up running everything in HTTP for some time. This would mean we loose that bit of telemetry, not ideal.
What are your thoughts on this?

Reply:
I have not, but let me check. Regarding the secure context we won't run into problems on localhost:

> Locally-delivered resources such as those with http://127.0.0.1 URLs, http://localhost and http://*.localhost URLs (e.g. http://dev.whatever.localhost/), and file:// URLs are also considered to have been delivered securely.

https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts

I think I can't make a decision for deployments. Do we have data about that?

Reply:
This is the simplest solution I could come up with:

```js
const text = "Hello Ivan!";

function bufferToHex(buffer) {
  return [...new Uint8Array(buffer)]
    .map((b) => b.toString(16).padStart(2, "0"))
    .join("");
}

async function digestMessage(message) {
  const msgUint8 = new TextEncoder().encode(message);
  const hashBuffer = await crypto.subtle.digest("SHA-256", msgUint8);
  return bufferToHex(hashBuffer);
}

(async () => {
  const hex = await digestMessage(text);
})();
```

I agree it looks a bit complex I agree, but on the other hand this can be isolated + tested well and if it helps to avoid adding ~6kB to the frontend-bundle I'd be willing to accept it.

Reply:
It's deployments specifically. I don't think we have any data on that, but judging by my experience (especially when I had not a lot of it), having deployments set up can be a pain, especially when you need SSL, etc. So using it in non-secure context (especially if the use-case allows) is not unheard of.

Reply:
Alright, what do you think about this pattern in FE:

```js
function bufferToHex(buffer) {
  return [...new Uint8Array(buffer)]
    .map((b) => b.toString(16).padStart(2, "0"))
    .join("");
}

async function digestMessage(message) {
  const msgUint8 = new TextEncoder().encode(message);
  const hashBuffer = await crypto.subtle.digest("SHA-256", msgUint8);
  return bufferToHex(hashBuffer);
}

const hash = require('hash.js');

const hashAdminUserEmail = async (payload) => {
  try {
    return await digestMessage(payload.email);
  } catch(error) {
    // not a secure context

    const hash = import('hash.js');
    return hash.sha256().update(payload.email).digest('hex');
  }
```

I'm not sure which type of error would be thrown in case of a non-secure context, but that's something we could find out I guess.

This would allow us to not increase the size of the main bundle and only load the dependency if it is needed.

Reply:
We could try this, looks good! Thank you kindly 😊

Reply:
Cool outcome! We discussed this a bit [here](https://github.com/strapi/strapi/pull/14083#discussion_r962926271) also.
The thrown error will be a `TypeError` because `crypto.subtle` is gonna be `undefined`.